src/containerapp/azext_containerapp/custom.py

```diff
@@ -1061,12 +1061,12 @@ def containerapp_up(cmd,
     env = ContainerAppEnvironment(cmd, managed_env, resource_group, location=location, logs_key=logs_key, logs_customer_id=logs_customer_id)
     app = ContainerApp(cmd, name, resource_group, None, image, env, target_port, registry_server, registry_user, registry_pass, env_vars, workload_profile_name, ingress)
 
-    _set_up_defaults(cmd, name, resource_group_name, logs_customer_id, location, resource_group, env, app)
-
     if app.check_exists():
         if app.get()["properties"]["provisioningState"] == "InProgress":
             raise ValidationError("Containerapp has an existing provisioning in progress. Please wait until provisioning has completed and rerun the command.")
 
+    _set_up_defaults(cmd, name, resource_group_name, logs_customer_id, location, resource_group, env, app)
```

Comment:
Why move the `_set_up_defaults` behind the `if app.check_exists()`?

Reply:
I think this fix no need to move the `_set_up_defaults` behind the `if app.check_exists()`, right?

Reply:
Without execute the `_set_up_defaults` first, maybe we cannot find the containerapp we want to match.

Reply:
Updated